components/eamxx/docs/user/diags/conditional_sampling.md

```diff
@@ -0,0 +1,88 @@
+# Conditional sampling
+
+The conditional sampling diagnostic allows you to extract field values
+where a specified condition is met, filling other locations with a
+default fill value. This is useful for analyzing field behavior under
+specific atmospheric conditions or at particular levels.
+The goal of the diagnostics is to enable you to discard unwanted data
+via *other* reductions, such as horizontal and vertical reductions
+(see "Field contraction" section of the docs).
+
+## Overview
+
+Conditional sampling creates a new field where:
+
+- Values from the input field are preserved
+  where the condition is **true**
+- Fill values (default: large number) are used
+  where the condition is **false**
+
+The general syntax is: `<input>_where_<condition>_<operator>_<value>`
+
+## Supported operators
+
+| Operator | Aliases | Description |
+| -------- | ------- | ----------- |
+| `eq` | `==` | Equal to |
+| `ne` | `!=` | Not equal to |
+| `gt` | `>` | Greater than |
+| `ge` | `>=` | Greater than or equal to |
+| `lt` | `<` | Less than |
+| `le` | `<=` | Less than or equal to |
+
+## Field-based conditional sampling
+
+Compare the input field against another field at each grid point.
+
+**Examples**:
+
+- `T_mid_where_qv_gt_0.01`
+- `p_mid_where_T_mid_le_273.15`
+- `qv_where_p_mid_lt_50000`
```

Comment:
Do we want to offer unit support?  For example for pressure slices we accept 50000Pa, 500hPa, 500mb

Reply:
I also vote for no units. Otherwise users may start to specify units for every var. What to do then? Verifying they match what eamxx uses is hard. Even harder if you want to support different units (like m and km or g and kg..). So yeah, users are responsible for knowing what units the fields have (can look up older output files metadata) and specify the numerical value accordingly.